include/claws/array_ops.hpp

```diff
@@ -170,5 +170,20 @@ namespace claws
     SCALAR_ARRAY_OPERATOR_DEF(>>);
   }
 
+  namespace impl
+  {
+    template<class Func, class T, std::size_t count, std::size_t... indices>
+    std::array<T, count> init_array(Func functor, std::index_sequence<indices...>)
```

Comment:
You should take Func as a Forwarding Reference, and the function must be noexcept if the functor is noexcept